Packager/Factories/CodingHistory/DatCodingHistoryGenerator.cs

```diff
@@ -0,0 +1,69 @@
+    {
+        private const string Line1Format = "A={0},M={1},T={2},";
+        private const string Line2Format = "A=PCM,F={0},W={1},M={2},T={3};A/D,";
+        private const string Line3Format = "A=PCM,F={0},W={1},M={2},T=Lynx AES16;DIO";
+
+        private const string ThirtyTwoK = "32k";
+        private const string FortyFourOneK = "44.1k";
+        private const string FortyEightK = "48k";
+        private const string NinetySixK = "96k";
+        private const string TwentyFourBit = "24";
+        private const string SixteenBit = "16";
+        private IDictionary<string, string> SampleRatesToText = new Dictionary<string, string>()
+        {
+            { ThirtyTwoK, "32000" },
+            { FortyFourOneK, "44100" },
+            { FortyEightK, "48000" },
+            { NinetySixK, "96000" }
+        };
+            
+        protected override string GenerateLine1(AudioPodMetadata metadata, DigitalAudioFile provenance, AbstractFile model)
+        {
+            //AssertSoundFieldSpecifiedInMetadata(metadata.SoundField);
+            var soundField = model.IsPreservationIntermediateVersion() ? MonoSoundField : StereoSoundField;
+            return string.Format(Line1Format, DigitalFormat, soundField, GeneratePlayerTextField(metadata, provenance));
+        }
+
+        protected override string GenerateLine2(AudioPodMetadata metadata, DigitalAudioFile provenance, AbstractFile model)
+        {
+            var soundField = model.IsPreservationIntermediateVersion() ? MonoSoundField : StereoSoundField;
+            var bitDepth = provenance.SampleRate == NinetySixK ? TwentyFourBit : SixteenBit;
+            
+            return string.Format(Line2Format,ConvertSampleRate(provenance.SampleRate), bitDepth, soundField, GenerateAdTextField(provenance));
+        }
+
+        protected override string GenerateLine3(AudioPodMetadata metadata, DigitalAudioFile provenance, AbstractFile model)
+        {
+            var soundField = model.IsPreservationIntermediateVersion() ? MonoSoundField : StereoSoundField;
+            var bitDepth = provenance.SampleRate == NinetySixK ? TwentyFourBit : SixteenBit;
+            return string.Format(Line3Format, ConvertSampleRate(provenance.SampleRate), bitDepth, soundField);
+        }
+
+        protected override string GenerateLine4(AudioPodMetadata metadata, DigitalAudioFile provenance, AbstractFile model)
+        {
+            return string.Empty;
+        }
+
+        /// <summary>
+        /// POD occasionally refactors without maintaining case sensitivity in its attribute values. This method should take any case of a sample rate 
+        /// (48K vs 48k for instance) and look up determine the appropriate match (48000 in this example).
+        /// </summary>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        private string ConvertSampleRate(string value)
+        {
+            return SampleRatesToText[value.ToLower()];
```

Comment:
This will throw an exception if the key doesn't exist. I would check to see that the key exists and then if it doesn't throw an `EmbeddedMetadataException` with an appropriate error message. Otherwise if somehow an invalid value gets passed in, you'll have to unwind the generic exception to figure out what went wrong.

Also, if you define the dictionary as having case-insensitive comparer above, you can get rid of "ToLower".

Finally, it might also be worth throwing an explicit`EmbeddedMetadataException` exception if value is null or whitespace. So, first test to ensure that value is set, then test to ensure that it exists in the dictionary.

Reply:
Resolved https://github.com/IUMDPI/IUMediaHelperApps/pull/29/commits/9257104f1a655b089a7c7d44db46738ccada92f9